### ISSUE TITLE
All RQJobs failing as 'workshops_person.data_privacy_agreement does not exist'

### ISSUE THREAD

**Issue by elichad** [carpentries/amy#2417]
A call to `Person.data_privacy_agreement` is lurking somewhere in `amy/autoemails/actions.py` when we set `context["instructors"]` (e.g. lines 512 or 913). This is causing all the email actions to fail since v4.1 was released.

Full traceback visible in any failed RQJob, but doesn't have line breaks so I haven't included one here.

I'm looking into this, but could do with some guidance from @pbanaszkiewicz on how to test the fix.

**Comment by elichad** [comment]
I tried to re-create this locally, but the job ran successfully...

**Comment by elichad** [comment]
@pbanaszkiewicz I can't find anything in the codebase that looks like it's referencing `Person.data_privacy_agreement` directly. I found one [stackoverflow issue](https://stackoverflow.com/questions/68084705/django-migration-deleted-column-now-django-says-it-doesnt-exist) that suggests it might be a caching issue, but I don't know if that's likely here since we recreated our cache tables with v4.1. But I guess it must be something to do with the configuration because I can't replicate locally.
I haven't tried restarting the server as I'm not sure if there's any potential fallout I should be aware of, plus it's working time for our US colleagues.
Do you have any ideas on this?

**Comment by pbanaszkiewicz** [comment]
I assume this is related to incompatible `Person` objects: a fresh one doesn't have that field, whereas RQJobs having been pickled contain a version that does have this field.

I fixed this once. Unfortunately the fix was manual and tedious, as every RQJob scheduled in previous AMY release would have to be unpickled and then re-pickled with new Person object.

**Comment by pbanaszkiewicz** [comment]
I'm going to leave here my investigation notes
---

I investigated the failing email automation jobs: https://github.com/carpentries/amy/issues/2417

Originally I thought that, due to the removal of old-style consent fields from Person model, the issues were related to old pickled version of Person instances and the same instances being incompatible (ie. without these fields).


However now I was able to re-try two of the emails scheduled to be sent today, as well as some new emails created for a test event of my own.


I was also able to correctly, ie. without any exceptions, read properties of old email jobs.


So it all turned out to be old processes (rqworker and rqscheduler) running without refresh after the upgrade. This should become part of the deployment scripts.


Note: it's not that easy to reload configuration for supervisord, the tool used to control rqworker and rqscheduler processes. There's some confusion about its commands:
https://www.onurguzel.com/supervisord-restarting-and-reloading/
https://github.com/Supervisor/supervisor/issues/720
Apparently reread + update should be used, or at least stop + start.
